src_python/org_server.py

```diff
@@ -0,0 +1,269 @@
+                    self.client_ids[client_socket] = "new"
+
+                    logging.info('Accepted new connection at %s:%s',*client_address)
+                    self.send_model(client_socket)
+
+                else:
+
+                    message = self.receive(notified_socket)
+
+                    if message is False:
+                        self.sockets_list.remove(notified_socket)
+                        del self.clients[notified_socket]
+                    else:
+                        client_id = message['CLIENT_ID']
+                        weights = message['WEIGHTS']
+                        num_examples = message["NUM_EXAMPLES"]
+                        self.client_ids[notified_socket] = client_id
+                    
+                        logging.info('Recieved model from client-%s at %s:%s',client_id, *self.clients[notified_socket])
+                        self.update_model(weights,int(num_examples))
+
+            for notified_socket in exception_sockets:
+                self.sockets_list.remove(notified_socket)
+                del self.clients[notified_socket]
+
+        with open(FLAG_PATH,'w') as f:
+            f.write("STOP")
+
+if __name__ == "__main__":
+
+    from models.supervised import Model
+
+    arg_names = [
+        'org_id',
+        'num_clients',
+        'num_rounds',
+        'IP',
+        'port'
+        ]
+
+    args = dict(zip(arg_names, sys.argv[1:]))
+
+    logging.warning('####################################### New Training Session #######################################')
+    logging.info('Server started , org ID %s, number of clients %s, number of rounds %s',args['org_id'],args['num_clients'],args['num_rounds'])
+
+    if 'IP' not in args.keys()  or args['IP'] == 'localhost':
+        args['IP'] = socket.gethostname()
+
+    if 'PORT' not in args.keys():
+        args['port'] = 5050
+
+    print (args['IP'])
+    
+    model_weights = ""
+    while True:
+        time.sleep(5)
+
+        flag = "1"
+        with open(FLAG_PATH,'r') as f:
+            flag  = f.read()
+        
+        data = ""
+        logging.info(type(flag))
+        if str(flag).strip() == "0":
+            with open(WEIGHTS_PATH,'rb') as f:
+                data = f.read()
+
+            data = pickle.loads(data[HEADER_LENGTH:])
+            model_weights = data["WEIGHTS"]
+    
+            break
+
+    print (model_weights)
+    server = Server(model_weights=model_weights,org_id=args['org_id'],rounds=int(args['num_rounds']),num_clients=int(args['num_clients']),ip=args['IP'],port=int(args['port']))
+    
+    logging.info('Federated training started!')
+    start = timer()
+    server.run()
+    end = timer()
+
+    elapsed_time = end -start
+    logging.info('Federated training done!')
+    logging.info('Training report : Elapsed time %s seconds, graph ID %s, number of clients %s, number of rounds %s',elapsed_time,args['org_id'],args['num_clients'],args['num_rounds'])
```

Comment:
Add a newline at the end of the file

Reply:
FIxed in 1875888487e22c8e17cbadd8acc558b28d98ebf0

Reply:
Done